slack-sup/models/sup.rb

```diff
@@ -87,6 +88,14 @@ def calendar_href(dt = nil)
 
   private
 
+  def select_best_captain(users)
+    users.map { |u| [u, last_time_captain(u)] }.to_h.min_by { |_k, v| v.to_i }.first
```

Comment:
In Ruby you have to think in English

* `get` instead of `select` is more idiomatic Ruby
* you already have users, so this is just a property on `Sup`
* users know when they were last captain, it's not Sup's responsibility

so this should be:

```ruby
def get_best_captain
   users.sort_by { |u| u.last_captain_at }.last   
end
```

Reply:
awesome suggestions! all make total sense! i was originally thinking of one mongo query to get `last_captain_at` for all users in one query. But this is LOT cleaner.